src/main/java/teammates/ui/controller/InstructorCourseAddAction.java

```diff
@@ -91,7 +91,7 @@ private void createCourse(CourseAttributes course) {
 
             InstructorAttributes curInstructor = logic.getInstructorForGoogleId(course.id, account.googleId);
 
-            if (Logic.isCourseArchived(course.id, curInstructor.googleId)) {
+            if (curInstructor.isArchived) {
```

Comment:
How about getter methods instead of accessing attributes directly, in case we declare them as private later